velox/expression/CastExpr-inl.h

```diff
@@ -275,6 +275,30 @@ VectorPtr CastExpr::applyDecimalToIntegralCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToBooleanCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& /*fromType*/,
+    const TypePtr& toType) {
+  VectorPtr result;
+  context.ensureWritable(rows, toType, result);
+  (*result).clearNulls(rows);
+  auto resultBuffer =
+      result->asUnchecked<FlatVector<bool>>()->mutableRawValues<uint64_t>();
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
+    auto value = simpleInput->valueAt(row);
+    if (value == 0) {
```

Comment:
nit: bits::setBit(resultBuffer, row, value != 0) ;

Reply:
Thanks, Updated it.